File: Microservice/Payment-Service/src/main/java/com/example/paymentservice/entity/Payment.java
package com.example.paymentservice.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.ZonedDateTime;

@Entity
@Table(name = "payment")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Payment {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long orderId;
    private BigDecimal amount;
    private BigDecimal paymentFee;
    private String paymentMethod; // COD, BANKING, PAYPAL, VNPAY
    private String  paymentStatus;
//    PENDING,
//    COMPLETED,
//    CANCELLED
    @CreationTimestamp
    private ZonedDateTime createdOn;
    @UpdateTimestamp
    private ZonedDateTime lastModifiedOn;



}

File: Microservice/Payment-Service/models/Payment.js
const mongoose = require('mongoose');

// Define the schema
const paymentSchema = new mongoose.Schema({
    orderid: { type: String, required: true },
    paymentMethod: { type: String, enum: ['COD', 'PAYPAL', 'VNPAY'], required: true },
    paymentStatus: { type: String, enum: ['PENDING', 'COMPLETED', 'CANCELLED'], required: true },
    vnpPayDate: {type:String},
    vnpTxnRef: {type:String},
    createdOn: { type: Date, default: Date.now },
    lastModifiedOn: { type: Date, default: Date.now }
});

// Create the model
const Payment = mongoose.model('Payment', paymentSchema);

module.exports = Payment;
